pkg/auth/auth_handler_test.go

```diff
@@ -0,0 +1,273 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+
+	"github.com/openshift/assisted-service/client"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
+	clientInstaller "github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/restapi"
+	"github.com/openshift/assisted-service/restapi/operations/installer"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func serv(server *http.Server) {
+	_ = server.ListenAndServe()
+}
+
+func GetTokenAndCert() (string, []byte) {
+
+	//Generate RSA Keypair
+	pub, priv, _ := GenKeys(2048)
+
+	//Generate keys in JWK format
+	pubJSJWKS, _, kid, _ := GenJSJWKS(priv, pub)
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.MapClaims{
+		"account_number": "1234567",
+		"is_internal":    false,
+		"is_active":      true,
+		"account_id":     "7654321",
+		"org_id":         "1010101",
+		"last_name":      "Doe",
+		"type":           "User",
+		"locale":         "en_US",
+		"first_name":     "John",
+		"email":          "jdoe123@example.com",
+		"username":       "jdoe123@example.com",
+		"is_org_admin":   false,
+		"clientId":       "1234",
+	})
+	token.Header["kid"] = kid
+	tokenString, _ := token.SignedString(priv)
+	return tokenString, pubJSJWKS
+}
+
+func TestAuth(t *testing.T) {
+	log := logrus.New()
+
+	userToken, JwkCert := GetTokenAndCert()
+	agentKeyValue := "fake_pull_secret"
+	userKeyValue := "bearer " + userToken
+	t.Parallel()
+	tests := []struct {
+		name                   string
+		expectedTokenValue     string
+		authInfo               runtime.ClientAuthInfoWriter
+		isListOperation        bool
+		enableAuth             bool
+		addHeaders             bool
+		expectedRequestSuccess bool
+	}{
+		{
+			name:                   "User Successful Authentication",
+			expectedTokenValue:     userKeyValue,
+			authInfo:               UserAuthHeaderWriter(userKeyValue),
+			isListOperation:        true,
+			enableAuth:             true,
+			addHeaders:             true,
+			expectedRequestSuccess: true,
+		},
+		{
+			name:                   "Fail auth without headers",
+			expectedTokenValue:     agentKeyValue,
```

Comment:
Need to test successful user and agent auth
Also unsuccessful user auth (for now all agent auth should be succesfull, until @rollandf adds his code)
also need to test case where there are no headers supplied for either agent or user auth

Reply:
Done.
As requested, added more unit tests in [PR-70](https://github.com/openshift/assisted-service/pull/70/files) `pkg/auth/auth_handler_test.go `